### PR TITLE
Fixes for invalid projects throwing 500 errors, and redirect loops ba…

### PR DESCRIPTION
**Pull request by Matthewrbowker**
…sed on default_project.  Also includes other minor fixes.

T166332
T166438

### CODE REVIEW

File: src/AppBundle/Controller/ArticleInfoController.php [hunk 1]
```diff
@@ -76,9 +77,14 @@ public function resultAction(Request $request)
...
         $page = $request->attributes->get('article');
 
-        // sets the database name within $this->lh
-        $dbValues = $this->lh->databasePrepare($project, 'ArticleInfo');
-        $projectUrl = $dbValues['url'];
+        $projectData = ProjectRepository::getProject($project, $this->container);
+
+        if (!$projectData->exists()) {
+            $this->addFlash("notice", ["invalid-project", $project]);
+            return $this->redirectToRoute("articleInfo");
```
Comment (samwilson) [inline src/AppBundle/Controller/ArticleInfoController.php]: I don't think we should redirect in GET requests like this, as it breaks navigation unnecessarily. Instead, if a project doesn't exist, the user should be shown the same form with all their values, as well as an error message, so they can more easily fix whatever typo they made.

File: src/AppBundle/Controller/ArticleInfoController.php [hunk 2]
```diff
@@ -76,9 +77,14 @@ public function resultAction(Request $request)
         $project = $request->attributes->get('project');
         $page = $request->attributes->get('article');
 
-        // sets the database name within $this->lh
-        $dbValues = $this->lh->databasePrepare($project, 'ArticleInfo');
-        $projectUrl = $dbValues['url'];
+        $projectData = ProjectRepository::getProject($project, $this->container);
```
Comment (samwilson) [inline src/AppBundle/Controller/ArticleInfoController.php]: Personally, I'd just call this `$project` as that's what it is.

File: src/AppBundle/Controller/PagesController.php [hunk 3]
```diff
@@ -61,20 +62,15 @@ public function indexAction($project = null)
 
         // set default wiki so we can populate the namespace selector
         if (!$project) {
-            $project = $this->container->getParameter('default_project');
+            $project = $this->getParameter('default_project');
```
Comment (samwilson) [inline src/AppBundle/Controller/PagesController.php]: Could also do `$project = ProjectRepository::getDefaultProject($this->container);`

File: src/AppBundle/Controller/TopEditsController.php [hunk 4]
```diff
@@ -65,11 +65,22 @@ public function indexAction(Request $request)
...
+            $namespaces = $project->getNamespaces();
+        }
+
         return $this->render('topedits/index.html.twig', [
             'xtPageTitle' => 'tool-topedits',
             'xtSubtitle' => 'tool-topedits-desc',
             'xtPage' => 'topedits',
-            'project' => $project,
+            'domain' => $domain,
+            'namespaces' => $namespaces,
```
Comment (samwilson) [inline src/AppBundle/Controller/TopEditsController.php]: In all of these controllers, I don't think it's necessary to pass `domain` and `namespaces` in to the view when `project` is already there and can provide these. Perhaps the form macros need to be more forgiving in what they receive.

File: src/AppBundle/Controller/ArticleInfoController.php (hunk 2 above)
Comment (Matthewrbowker) [inline src/AppBundle/Controller/ArticleInfoController.php]: We really need to get the variable names uniform - in this case, $project is in use elsewhere.

File: src/AppBundle/Controller/ArticleInfoController.php (hunk 1 above)
Comment (Matthewrbowker) [inline src/AppBundle/Controller/ArticleInfoController.php]: Indeed.  I wonder if we can make our forms smart enough that we can post back and have the form display the values that way.  I'm not 100% sure though...

File: src/AppBundle/Controller/PagesController.php (hunk 3 above)
Comment (Matthewrbowker) [inline src/AppBundle/Controller/PagesController.php]: Huh, didn't know that.  Thanks.

File: src/AppBundle/Controller/TopEditsController.php (hunk 4 above)
Comment (Matthewrbowker) [inline src/AppBundle/Controller/TopEditsController.php]: Indeed.  But in this case, it also allows us to set $domain and $namespaces to default values - twig templates don't have very good error handling from what I can tell.

File: src/AppBundle/Controller/SimpleEditCounterController.php [hunk 5]
```diff
@@ -57,11 +58,15 @@ public function resultAction($project, $username)
...
-        $dbName = $dbValues["dbName"];
-        $wikiName = $dbValues["wikiName"];
-        $url = $dbValues["url"];
+        if (!$projectData->exists()) {
+            $this->addFlash("notice", ["invalid-project", $project]);
+            return $this->redirectToRoute("SimpleEditCounter");
+        }
+
+        $dbName = $projectData->getDatabaseName();
+        $url = $projectData->getUrl();
```
Comment (MusikAnimal) [inline src/AppBundle/Controller/SimpleEditCounterController.php]: Speaking of variable names, I think this one should be `$projectUrl` (like it is in some other places), to differentiate from other URLs.

File: src/AppBundle/Controller/AutomatedEditsController.php [hunk 6]
```diff
@@ -67,21 +68,30 @@ public function indexAction(Request $request)
...
+        }
 
-        // TODO: add namespace support
         return $this->render('autoEdits/index.html.twig', [
             'xtPageTitle' => 'tool-autoedits',
             'xtSubtitle' => 'tool-autoedits-desc',
             'xtPage' => 'autoedits',
             'project' => $project,
-            // 'namespaces' => $api->namespaces($project),
+            'namespaces' => $namespaces,
```
Comment (MusikAnimal) [inline src/AppBundle/Controller/AutomatedEditsController.php]: I don't believe per-namespace support is there yet but I'm going to add it soon, so no harm in uncommenting

Comment (Matthewrbowker) [inline src/AppBundle/Controller/AutomatedEditsController.php]: I know, I figured it's not a big deal to fix it before it becomes an issue.

**Comment by Matthewrbowker** [comment]
Excellent.  I will leave on my to-do list to go through and redo all of the variable names, it will be a seprate PR though.  I'm merging this one since it has sign-off.

**Pull request merged**